lnst/Recipes/ENRT/SRIOVNetnsRecipe.py

```diff
@@ -0,0 +1,158 @@
+from lnst.Common.Parameters import Param
+from lnst.Common.IpAddress import ipaddress
+from lnst.Controller import HostReq, DeviceReq, RecipeParam
+from lnst.Controller.NetNamespace import NetNamespace
+from lnst.Recipes.ENRT.BaremetalEnrtRecipe import BaremetalEnrtRecipe
+from lnst.RecipeCommon.Ping.PingEndpoints import PingEndpoints
+from lnst.Recipes.ENRT.ConfigMixins.OffloadSubConfigMixin import (
+    OffloadSubConfigMixin,
+)
+from lnst.Recipes.ENRT.ConfigMixins.CommonHWSubConfigMixin import (
+    CommonHWSubConfigMixin,
+)
+
+
+class SRIOVNetnsRecipe(
+    CommonHWSubConfigMixin, OffloadSubConfigMixin, BaremetalEnrtRecipe
+):
+    """
+    This recipe implements Enrt testing for a simple SRIOV network scenario
+    with VF located in the network namespace that looks as follows
+
+    .. code-block:: none
+
+                    +--------+
+             +------+ switch +-----+
+             |      +--------+     |
+          +--+-+                 +-+--+
+    +-----|eth0|-----+     +-----|eth0|-----+
+    |     +----+     |     |     +----+     |
+    |       |        |     |       |        |
+    |  +-namespace-+ |     |  +-namespace-+ |
+    | |    vf0     | |     | |    vf0     | |
+    | +-----------+  |     | +-----------+  |
+    |     host1      |     |     host2      |
+    +----------------+     +----------------+
+
+    All sub configurations are included via Mixin classes.
+
+    The actual test machinery is implemented in the :any:`BaseEnrtRecipe` class.
+    """
+    host1 = HostReq()
+    host1.eth0 = DeviceReq(label="net1", driver=RecipeParam("driver"))
+
+    host2 = HostReq()
+    host2.eth0 = DeviceReq(label="net1", driver=RecipeParam("driver"))
+
+    offload_combinations = Param(default=(
+        dict(gro="on", gso="on", tso="on", tx="on", rx="on"),
+        dict(gro="off", gso="on", tso="on", tx="on", rx="on"),
+        dict(gro="on", gso="off", tso="off", tx="on", rx="on"),
+        dict(gro="on", gso="on", tso="off", tx="off", rx="on"),
+        dict(gro="on", gso="on", tso="on", tx="on", rx="off")))
+
+    def test_wide_configuration(self):
+        """
+        Test wide configuration for this recipe involves just adding an IPv4 and
+        IPv6 address to the matched eth0 nics on both hosts.
```

Comment:
is the docstring accurate or copied from a different recipe?

I would probably add information about the vf creation and it moving into a namespace...

Reply:
Fixed.